src/tests/ingestion.int.test.js

```diff
@@ -0,0 +1,45 @@
+// Mocked import
+import childProcess from 'child_process' // eslint-disable-line import/order
+
+// Test imports
+import path from 'path'
+import { PassThrough } from 'stream'
+import FileIngestionEngine from '%src/components/ingestion/FileIngestionEngine'
+
+jest.mock('child_process')
+
+describe('ingestion', () => {
+	describe('FileIngestionEngine', () => {
+		it('it should read a video file into a Kafka topic', (done) => {
+			jest.clearAllMocks()
+
+			const mockDate = new Date(1592156234000)
+			jest
+				.spyOn(global, 'Date')
+				.mockImplementation(() => mockDate)
+
+			const ingestionEngine = new FileIngestionEngine(
+				path.join(__dirname, '/data/norm.ts'),
+			)
+
+			const stream = new PassThrough()
+			childProcess.spawn.mockReturnValueOnce({
+				stdout: stream,
+				stdin: stream,
+				kill: jest.fn(),
+			})
+			ingestionEngine.producer = {
+				send: jest.fn().mockResolvedValue(),
+				connect: jest.fn().mockResolvedValue(),
+				disconnect: jest.fn(),
+			}
+
+			ingestionEngine.start()
+
+			ingestionEngine.producer.disconnect.mockImplementation(() => {
+				expect(ingestionEngine.producer.send).toMatchSnapshot()
```

Comment:
Ty for adding this to the test!

Would it be too far to *additionally* (or maybe alternatively, depending on what you think after reading this) re-combine all of the buffers of the resulting calls, and compare that buffer to the original video?  Basically, we want to make sure that the original video was sent in order and that the resulting payload stream reconstructs that video.

I am worried that it will break regularly as we change what Payloads look like.

That's fine (re-snapshotting is just a thing we will be doing), but I am also worried that the circumstances that re-snapshotting happen are also the ones where we are relying on tests to ensure nothing broke (e.g. the re-snapshot will pass, but the new snapshot might be broken without us realizing).

Reply:
Ah, I see what you're saying. That feels like a lot of logic for a test, and snapshots seem like a really clean way to deal with this, but maybe there's a short cut way to do that. Will think on it.

Reply:
Did this.